src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -47,10 +54,14 @@ public Result charTyped(char charTyped,
                             @NotNull Project project,
                             @NotNull Editor editor,
                             @NotNull PsiFile file) {
-        LSPFormattingFeature formattingFeature = getClientConfigurationSettings(file);
+        LSPFormattingFeature formattingFeature = getFormattingFeature(file);
         if (formattingFeature != null) {
+            boolean rangeFormattingSupported = formattingFeature.isRangeFormattingSupported(file);
+
             // Close braces
-            if (formattingFeature.isFormatOnCloseBrace(file)) {
+            if (formattingFeature.isFormatOnCloseBrace(file) &&
+                // Make sure the formatter supports formatting of the configured scope
+                ((formattingFeature.getFormatOnCloseBraceScope(file) == FormattingScope.FILE) || rangeFormattingSupported)) {
```

Comment:
In each case, we confirm that the configured scope is supported by the language server's formatter before even trying to format. As stated elsewhere, for language server's that don't support range formatting -- or support it pooly, e.g., the CSS language server -- the user can still enable formatting on close brace and/or statement terminator and have it format the entire file if desired.